### PR TITLE
Issue 17899 string interning

### PR DESCRIPTION
**Pull request by wezell**
1. I added the sync/double null check on the singleton as I saw multiple threads getting creating new  `INSTANCE`s when debugging

2. Using reflection to instantiate the JsonFactory and then modify the config value. 

#### How I tested
In my debugging, I saw (I think) the value change.  I had a copy of the JsonXContent class in my branch and added breakpoints when it was called.  This is the value from the jsonFactory before I set `INTERN_FIELD_NAMES`
### Before being set:
Screen Shot 2020-01-24 at 11 07 03 AM
### After being set, when running
This is the jsonFactory when running  `INTERN_FIELD_NAMES`
Screen Shot 2020-01-24 at 11 08 09 AM

Additionally I added a breakpoint on java's String.intern() method and did not see it get hit during elasticsearch use.   Thumbs up I think.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java [hunk 1]
```diff
@@ -15,25 +19,64 @@
...
+     * 
+     * @return
+     */
+    private static boolean disableStringIntern() {
+        try {
+            // init the static field
+            XContentBuilder x = JsonXContent.contentBuilder();
+
+            // get a handle on the field
+            Field jsonFactoryField = JsonXContent.class.getDeclaredField("jsonFactory");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Local variable 'jsonFactoryField' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java [hunk 2]
```diff
@@ -15,25 +19,64 @@
...
+     * instance. If this method fails for any reason, it is not tragic, just not optimal. Also, this is
+     * hard to test for - the only indication that any change has happened to the JsonFactory is an
+     * inner flag on the JsonFactory object itself
+     * 
+     * @return
+     */
+    private static boolean disableStringIntern() {
+        try {
+            // init the static field
+            XContentBuilder x = JsonXContent.contentBuilder();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Local variable 'x' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java [hunk 3]
```diff
@@ -15,25 +19,64 @@
...
+
+            // get a handle on the field
+            Field jsonFactoryField = JsonXContent.class.getDeclaredField("jsonFactory");
+            jsonFactoryField.setAccessible(true);
+            JsonFactory jsonFactory = (JsonFactory) jsonFactoryField.get(null);
+
+            // set the value
+            Logger.info(RestHighLevelClientProvider.class, "disabling String.intern() in Jackson");
+            jsonFactory.configure(JsonFactory.Feature.INTERN_FIELD_NAMES, false);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Avoid variables with short names like x](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java [hunk 4]
```diff
@@ -15,25 +19,64 @@
...
+     */
+    private static boolean disableStringIntern() {
+        try {
+            // init the static field
+            XContentBuilder x = JsonXContent.contentBuilder();
+
+            // get a handle on the field
+            Field jsonFactoryField = JsonXContent.class.getDeclaredField("jsonFactory");
+            jsonFactoryField.setAccessible(true);
+            JsonFactory jsonFactory = (JsonFactory) jsonFactoryField.get(null);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Local variable 'jsonFactory' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java]: Codacy Issue found: [Avoid unused local variables such as 'x'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java (hunk 2 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java:59]: yeah x is not a good name

**Review by jdotcms — APPROVED** [review]

**Pull request merged**